pandas/io/formats/style.py

```diff
@@ -561,11 +561,15 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             Whitespace shouldn't matter and the final trailing ';' shouldn't
             matter.
         """
-        for row_label, v in attrs.iterrows():
-            for col_label, col in v.items():
-                i = self.index.get_indexer([row_label])[0]
-                j = self.columns.get_indexer([col_label])[0]
-                for pair in col.rstrip(";").split(";"):
+        rows = [(row_label, v) for row_label, v in attrs.iterrows()]
```

Comment:
This can have pretty non-trivial memory impact for larger data frames. How much of a speed up are you seeing? Have you profiled the memory usage as well?

Reply:
No I have not. How do I do memory/speed profiling?

Reply:
ASV result attached in the conversation